static/js/addons-config/array-field.js

```diff
@@ -0,0 +1,406 @@
+/**
+ * Input Field for JSON-schema type:array.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * 
+ * This Source Code includes react-jsonschema-form
+ * released under the Apache License 2.0.
+ * https://github.com/mozilla-services/react-jsonschema-form
+ * Date on whitch referred: Thu, Mar 08, 2018  1:08:52 PM
+ */
+
+'use strict';
+
+/* globals SchemaUtils, SchemaField, Utils, UnsupportedField */
+
+function ArrayField(
+  schema,
+  formData,
+  idSchema,
+  name,
+  definitions,
+  onChange,
+  required = false,
+  disabled = false,
+  readonly = false) {
+
+  this.schema = SchemaUtils.retrieveSchema(schema, definitions);
+  this.formData = formData || [];
+  this.idSchema = idSchema;
+  this.name = name;
+  this.definitions = definitions;
+  this.onChange = onChange;
+  this.required = required;
+  this.disabled = disabled;
+  this.readonly = readonly;
+
+  return this;
+}
+
+ArrayField.prototype.require = function (name) {
+  return (
+    Array.isArray(this.schema.required) &&
+    this.schema.required.indexOf(name) !== -1
+  );
+};
+
+ArrayField.prototype.onChangeForIndex = function (index) {
+  return function (value) {
+    const newFormData = this.formData.map((item, i) => {
+      // We need to treat undefined items as nulls to have validation.
+      // See https://github.com/tdegrunt/jsonschema/issues/206
+      const jsonValue = typeof value === 'undefined' ? null : value;
+      return index === i ? jsonValue : item;
+    });
+
+    this.formData = newFormData;
+
+    if (this.onChange) {
+      this.onChange(this.formData);
+    }
+  }.bind(this);
+};
+
+ArrayField.prototype.onSelectChange = function (value, all) {
+  return function (event) {
+    const checked = event.target.checked;
+
+    if (checked) {
+      this.formData.push(value);
+      this.formData.sort((a, b) => all.indexOf(a) > all.indexOf(b));
+    } else {
+      this.formData = this.formData.filter(v => v !== value);
+    }
+
+    if (this.onChange) {
+      this.onChange(this.formData);
+    }
+  }.bind(this);
+};
+
+ArrayField.prototype.allowAdditionalItems = function () {
+  const schema = this.schema;
+  if (schema.additionalItems === true) {
+    console.warn('additionalItems=true is currently not supported');
+  }
+  return SchemaUtils.isObject(schema.additionalItems);
+};
+
+ArrayField.prototype.isAddable = function (formItems) {
+  const schema = this.schema;
+  let addable = true;
+
+  if (typeof schema.maxItems !== 'undefined') {
+    addable = addable && formItems.length < schema.maxItems;
```

Comment:
There's no need to do `addable &&` here, since addable is automatically true at this point.